core/handler/manager_server.go

```diff
@@ -103,8 +105,8 @@ func (h *handlerManager) GetDevice(ctx context.Context, in *pb.DeviceIdentifier)
 }
 
 func (h *handlerManager) SetDevice(ctx context.Context, in *pb.Device) (*empty.Empty, error) {
-	if !in.Validate() {
-		return nil, errors.BuildGRPCError(errors.NewErrInvalidArgument("Device", "validation failed"))
+	if err := in.Validate(); err != nil {
+		return nil, errors.BuildGRPCError(errors.NewErrInvalidArgument("Device", err.Error()))
```

Comment:
Be consistent with `errors.Wrap` vs wrapping it in `errors.NewErrInvalidArgument`